### PR TITLE
show VideoInfo loaded state in Compose previews

### PR DESCRIPTION
**Pull request by maxhille**
Just to get a bit more into Compose and improve the previews

### CODE REVIEW

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt [hunk 1]
```diff
@@ -29,12 +29,7 @@ import androidx.compose.foundation.IndicationInstance
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.interaction.InteractionSource
 import androidx.compose.foundation.interaction.MutableInteractionSource
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.layout.*
```
Comment (strooooke) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt:32]: Now that's why I believe that autoformat config belongs with the project, even though it's IntelliJ-specific.

But apart from that, autoconfig or not, I think *-imports are bad in general.

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt [hunk 2]
```diff
@@ -24,11 +24,7 @@ package com.lambdasoup.watchlater.ui.add
 
 import androidx.compose.animation.AnimatedVisibility
 import androidx.compose.foundation.Image
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
-import androidx.compose.foundation.layout.height
-import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.*
```
Comment (strooooke) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt:27]: As above.

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt [hunk 3]
```diff
@@ -226,3 +217,44 @@ fun AddScreenPreview() = AddScreen(
         )
     )
 )
+
+@Preview(name = "deutsch", locale = "de")
+@Preview(name = "english", locale = "en")
+@Preview(name = "night", uiMode = UI_MODE_NIGHT_YES)
+@Composable
+fun AddScreenPreviewInfoLoaded() = AddScreen(
```
Comment (strooooke) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt:225]: No interest in using `@PreviewParameter`?

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt [hunk 4]
```diff
@@ -159,11 +165,43 @@ private fun errorText(
         is VideoInfo.ErrorType.Youtube -> when (errorType.error) {
             YoutubeRepository.ErrorType.Network -> stringResource(id = R.string.error_network)
             YoutubeRepository.ErrorType.VideoNotFound -> stringResource(id = R.string.error_video_not_found)
-            else -> stringResource(id = R.string.could_not_load,
-                                   errorType.toString())
+            else -> stringResource(
```
Comment (strooooke) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt:168]: :+1:

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt [hunk 5]
```diff
@@ -159,11 +165,43 @@ private fun errorText(
...
+@Composable
+fun VideoSnippetPreviewLoaded() = VideoSnippet(
+    videoInfo = VideoInfo.Loaded(
+        data = YoutubeRepository.Videos.Item(
+            id = "video-id",
+            snippet = YoutubeRepository.Videos.Item.Snippet(
+                title = "Video Title",
+                description = "Video description",
+                thumbnails = YoutubeRepository.Videos.Item.Snippet.Thumbnails(
+                    medium = YoutubeRepository.Videos.Item.Snippet.Thumbnails.Thumbnail("dummy-url ignore in preview"),
```
Comment (strooooke) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/VideoSnippet.kt:201]: I wish we could find a good place to keep the preview fixtures around; they start to repeat.

**Review by strooooke — CHANGES_REQUESTED** [review]
Lovely previews!

**Comment by strooooke** [comment]
Excellent. So to check locally, I'd go `./gradlew ktlintCheck`? Could you please add that to the README?

**Review by strooooke — APPROVED** [review]

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt (hunk 3 above)
Comment (maxhille) [reply app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt]: Done

**Pull request merged**